File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task182.py

def clean_phone_number(phone_number):
    if phone_number is None:
        return None
    
    # Remove all non-digit characters
    digits_only = ''.join(filter(str.isdigit, phone_number))
    
    # Check if it starts with country code 1 and has 11 digits
    if len(digits_only) == 11 and digits_only.startswith('1'):
        digits_only = digits_only[1:]
    
    # Validate the cleaned number
    if len(digits_only) != 10:
        return None
    
    # Validate area code (first digit must be 2-9)
    if digits_only[0] < '2' or digits_only[0] > '9':
        return None
    
    # Validate exchange code (first digit must be 2-9)
    if digits_only[3] < '2' or digits_only[3] > '9':
        return None
    
    return digits_only

if __name__ == "__main__":
    # Test cases
    print("Test 1:", clean_phone_number("+1 (613)-995-0253"))
    print("Test 2:", clean_phone_number("613-995-0253"))
    print("Test 3:", clean_phone_number("1 613 995 0253"))
    print("Test 4:", clean_phone_number("613.995.0253"))
    print("Test 5:", clean_phone_number("(234) 567-8901"))

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task182.c

#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

char* cleanPhoneNumber(const char* phoneNumber) {
    if (phoneNumber == NULL) {
        return NULL;
    }
    
    // Allocate memory for result
    char* result = (char*)malloc(11 * sizeof(char));
    if (result == NULL) {
        return NULL;
    }
    
    // Remove all non-digit characters
    char digitsOnly[20];
    int digitCount = 0;
    
    for (int i = 0; phoneNumber[i] != '\\0'; i++) {
        if (isdigit(phoneNumber[i])) {
            digitsOnly[digitCount++] = phoneNumber[i];
        }
    }
    digitsOnly[digitCount] = '\\0';
    
    // Check if it starts with country code 1 and has 11 digits
    int startIndex = 0;
    if (digitCount == 11 && digitsOnly[0] == '1') {
        startIndex = 1;
        digitCount--;
    }
    
    // Validate the cleaned number
    if (digitCount != 10) {
        free(result);
        return NULL;
    }
    
    // Validate area code (first digit must be 2-9)
    if (digitsOnly[startIndex] < '2' || digitsOnly[startIndex] > '9') {
        free(result);
        return NULL;
    }
    
    // Validate exchange code (first digit must be 2-9)
    if (digitsOnly[startIndex + 3] < '2' || digitsOnly[startIndex + 3] > '9') {
        free(result);
        return NULL;
    }
    
    // Copy the cleaned number to result
    strncpy(result, digitsOnly + startIndex, 10);
    result[10] = '\\0';
    
    return result;
}

int main() {
    // Test cases
    char* result1 = cleanPhoneNumber("+1 (613)-995-0253");
    printf("Test 1: %s\\n", result1 ? result1 : "NULL");
    free(result1);
    
    char* result2 = cleanPhoneNumber("613-995-0253");
    printf("Test 2: %s\\n", result2 ? result2 : "NULL");
    free(result2);
    
    char* result3 = cleanPhoneNumber("1 613 995 0253");
    printf("Test 3: %s\\n", result3 ? result3 : "NULL");
    free(result3);
    
    char* result4 = cleanPhoneNumber("613.995.0253");
    printf("Test 4: %s\\n", result4 ? result4 : "NULL");
    free(result4);
    
    char* result5 = cleanPhoneNumber("(234) 567-8901");
    printf("Test 5: %s\\n", result5 ? result5 : "NULL");
    free(result5);
    
    return 0;
}
